doc/Programming-guide/Key-platform-features/Data-collection/index.md

```diff
@@ -5,4 +5,1175 @@ permalink: /:path/
+by default for each Kaa application. This schema supports versioning, therefore, whenever a new log schema is configured on the Kaa service for the application,
+this new schema gets a new sequence version assigned. The Kaa service maintains compatibility with the older versions of the log schema to ensure proper
+functioning of the clients that for some reason are not yet upgraded to the latest schema version.
+
+The following examples illustrate basic log schemas.
+
+* The simplest definition of a log record with no data fields (mostly useless):
+
+```json
+{  
+    "name":"EmptyLog",  
+    "namespace":"org.kaaproject.sample",  
+    "type":"record",  
+    "fields":[
+    ]
+}
+```
+
+* A simple log schema with the log level, tag, and message:
+
+```json
+{
+    "name":"LogData",
+    "namespace":"org.kaaproject.sample",
+    "type":"record",
+    "fields":[
+        {
+            "name":"level",
+            "type":{
+                "type":"enum",
+                "name":"Level",
+                "symbols":[
+                    "DEBUG",
+                    "ERROR",
+                    "FATAL",
+                    "INFO",
+                    "TRACE",
+                    "WARN"
+                ]
+            }
+        },
+        {
+            "name":"tag",
+            "type":"string"
+        },
+        {
+            "name":"message",
+            "type":"string"
+        }
+    ]
+}
+```
+
+## Adding log schema
+
+The default log schema installed for Kaa applications is empty. You can configure your own log schema using the Admin UI or
+[Admin REST API]({{root_url}}Programming-guide/Server-REST-APIs/ #TODO).
+For the purpose of this guide, we will use schema that is very close to the common log structure:
+* log level
+* tag
+* message
+
+```json
+{
+    "type":"record",
+    "name":"LogData",
+    "namespace":"org.kaaproject.kaa.schema.sample.logging",
+    "fields":[
+        {
+            "name":"level",
+            "type":{
+                "type":"enum",
+                "name":"Level",
+                "symbols":[
+                    "DEBUG",
+                    "ERROR",
+                    "FATAL",
+                    "INFO",
+                    "TRACE",
+                    "WARN"
+                ]
+            }
+        },
+        {
+            "name":"tag",
+            "type":"string"
+        },
+        {
+            "name":"message",
+            "type":"string"
+        }
+    ]
+}
+```
+
+In Admin UI as a tenant developer, you can create new log schemas for the application as follows:
+
+1. In the Log schemas window for the application, click Add schema.
+2. In the Add log schema window, create a log schema either by using the [schema form]({{root_url}}LINK_TO_AVRO_UI_FORMS_AVRO_UI_SCHEMA_FORM) or
+by uploading a schema in the [Avro](http://avro.apache.org/docs/current/spec.html) format from a file.<br/>
+
+![Add log schema](attach/add-log-schema.png) <br/>
+
+3. Click Add to save the schema.
```

Comment:
At this point the enumeration is re-started (in compiled documentation) which is wrong.